### ISSUE TITLE
Surfacing actionable mail from VBMS document types

### ISSUE THREAD

**Issue by laurjpeterson** [department-of-veterans-affairs/caseflow#5950]
## Background

Veterans and appellants send in all different types of mail at any point in the appeals process. Mail is typically sent to the central mail processing center in Janesville, Wisconsin. Employees in Janesville upload the mail in the Mail Portal. If the mail is for BVA, BVA employees process it in the following way:
- Mail supervisor assigns mail to individual mail staff in the Mail Portal
- Mail employee opens the mail in the mail portal 
- Mail employee uploads the mail to VBMS
- Mail employee searches for the Veteran and adds a "mail tab" in VACOLS and assigns it based on routing rules

Mail comes in to:
- Board P.O. Box
- VBA P.O. Box

## Goals
- Surface actionable mail to the right employee as soon as possible
- Identify missed mail (e.g. that goes to VBA mail box) by recognizing that there is an open appeal

**Potential goal:**
- Increase mail team productivity?

## Approach/Strategy
- Rely on VBMS document ID for the type of mail (replacing VACOLS mail types).
- Add backend routing rules based on VBMS document ID so those documents are surfaced to the right user. 
- Add a mail task.
  - Certain document types are blocking if the case is at decision
  - Certain document types can be done in parallel

### Assumptions
- We can use VBMS document type to identify the type of document and base our routing on them. 
- VBMS document types are easily changed by VBA and BVA users, and users will know how to change them.

### Tech spec

https://github.com/department-of-veterans-affairs/caseflow/issues/6763

## Stories
- If there is an active appeal at the Board, and there is mail addressed to the **VBA mailbox**, the Board user should be [notified] that there is new mail for the Veteran.
- If there is an active appeal at the Board, and there is mail addressed to the **BVA mailbox**, the Board user should be [notified] that there is new mail for the Veteran.

- If the mail is of document type [x] in VBMS, it should be routed to [y].
   - Note: need to create this mapping.
   - Note: need to understand if [y] is a person or a place

## Questions
- How are all of these mail labels used? How similar are they to VBMS document types?
- How might we auto-assign mail or reduce the amount of mis-assigned mail?
- How might we simplify mail routing rules?
- How does mail that goes to a VBA P.O. Box get into VBMS? Does Janesville, Centralized Mail, upload it to VBMS? How do we differentiate this from mail that comes to BVA P.O. Box?
- If non-blocking mail tasks are added, how does the user who has the case currently view it?
- If a blocking mail task is added, what does the attorney or judge see?
   - On hold tab?

## Resources
- [Mail Mural](https://app.mural.co/t/workqueue2001/m/workqueue2001/1505315222173/4e3178cd7e6897c73729a40ef1e9c5d88fd403f1) - Discovery 2017
- [Updated Mail Assignment Sheet (routing rules) - August 2018](https://github.com/department-of-veterans-affairs/appeals-design-research/blob/master/Projects/Caseflow%20Queue/Research/Documents/Mail%20Assignment%20for%20August%202018%20(August%2020).doc)
- [Mail portal screenshot](https://github.com/department-of-veterans-affairs/appeals-design-research/blob/master/Projects/Caseflow%20Queue/Research/Documents/mail%20portal.png)

### Mail notes from convo 4/12/2018
- Mail Portal + electronic faxes
- Mail = VBA product
- Roles: user, super user, supervisor

#### Mail Portal
- Queues
	- Assignment All (someone coming and putting mail on my desk) - mail after upload in Jainsville, WI
	- Authorization All (super, supervisor)
	- Hold All (user) (super, supervisor)
	- Reassign All (super, supervisor)
	- UM First Auth (UM = unidentifiable mail) (supervisor)
	- Work All (user)(super, supervisor)
- Corey distributes work from the portal to users - assigns a user
- Goes into the mail person's "Work All" queue
- Want to work earliest mail received
- Search veteran's name in VACOLS
- Input mail in VACOLS
- Choose type of mail, auto-populates notes
- Choose next person related to routing rules
- When another VACOLS user opens appeal in VACOLS, there is a mail pop-up

### Types of Mail
From Jed, 06/19/2018
Field Name:	MLTYPE

VACOLS field 
01	Change of Address | Correspondence - Miscellaneous | Change of Address/Direct Deposit Request
02	Congressional Interest | Congressionals | Correspondence
03	Controlled Correspondence | *Need definition from BVA*
04	CUE Related | CUE Related | Appeals
05	Evidence or Argument | Buddy / Lay Statement<br>Medical Treatment Record - Government Facility<br>Medical Treatment Record - Non-Government Facility<br>Photographs<br>Web / HTML Documents<br>Bank/Financial Statement<br> | Non-Medical Evidence<br>Medical Records<br>Medical Records<br>Medical Records<br>Non-Medical Evidence<br>*STILL WORKING HERE*

06	FOIA Request | Correspondence | Freedom of information Act (FOIA) / Privacy Act Request
07	Hearing Related | Appeals | Hearing Request (not exact)
08	Motion to Advance on Docket
09	Motion for Reconsideration
10	Power of Attorney Related | Representation<br>Representation – National Service Organization<br>Representation – Other| Power of Attorney (incl. VA 21-22, VA 22a)
11	Professional Service Mail
12	Returned or Undeliverable Mail | Correspondence – Returned Mail | Return Mail
13	Status Inquiry
14	Death Certificate | Non-Medical Evidence – Death Evidence | Death Certificate
15	Appellate Group Mail
16	Extension Request
17	Privacy Act Request | Correspondence | Freedom of information Act (FOIA) / Privacy Act Request
18	Other Motion
19	Privacy Complaints
20	Intra/Inter Agency Request
21	Ebert Temporary Transfer
22	How Do I Appeal Pamphlet
23	ECA Revocation
24	FOIA Request & Other Actions
25	Motion to Vacate
26	Attorney Inquiry
27	Withdrawal of Appeal

**Comment by laurjpeterson** [comment]
@nicholasholtz and I briefly chatted about how this idea for mail, relying on VBMS document types, could relate to Caseflow Intake. cc: @abbyraskinUSDS @shanear @leahbannon @suzchap 

For the new NOD form: Centralized Mail adds new NOD form VBMS document type. Caseflow recognizes that document type and moves the case into Caseflow intake.

**Comment by laurjpeterson** [comment]
Idea:
- Rely on centralized mail intake to apply VBMS document type (already happens today)
- Recreate the routing rules:
   - For cases that are not assigned to attorneys, case management 
   - Or lit support... etc.

**Comment by laurjpeterson** [comment]
Tracking some research tasks in appeals-design-research, [here](https://github.com/department-of-veterans-affairs/appeals-design-research/issues/922#issuecomment-403944914).

**Comment by laurjpeterson** [comment]
<img width="676" alt="screen shot 2018-07-19 at 1 11 53 pm" src="https://user-images.githubusercontent.com/27704093/42959026-5ea4d39a-8b55-11e8-8903-5bc69a29a5a4.png">

**Comment by laurjpeterson** [comment]
Chris reviewed the VBMS SCD. 

> Looking through this, it doesn't look like we have where it was sent. `systemSource` gives us the system that uploaded it, but that would probably be the same for both VBA and the Board? The `sourceComment` field could potentially help if it was set, although it's not currently used for anything coming through Janesville.

[0003AE_eFolder v1.0 SCD 15.1[2].pdf](https://github.com/department-of-veterans-affairs/caseflow/files/2263540/0003AE_eFolder.v1.0.SCD.15.1.2.pdf)

**Comment by laurjpeterson** [comment]
@sneha-pai @amprokop @allyceh and @laurjpeterson met 8/20 to review mail murals.

Next steps:
1. understand the feasibility of creating mail tasks based VBMS document types - eng
2. based on feasibility, define/redefine mail routing rules - product, 
3. design receiving, processing, and closing mail task (user agnostic, but likely co-located + AOD users)
4. design mail management branch creating mail task (if needed)
5. general mail interviews with eugene (co-located) and sal (hearings)

**Comment by laurjpeterson** [comment]
[Updated Mail Assignment Sheet - August 2018](https://github.com/department-of-veterans-affairs/appeals-design-research/blob/master/Projects/Caseflow%20Queue/Research/Documents/Mail%20Assignment%20for%20August%202018%20(August%2020).doc)

**Comment by allyceh** [comment]
@laurjpeterson @lowellrex @mdbenjam 

Open questions that will help us define backend requirements: 

- Does the mail team ever create multiple mail tasks in VACOLS if they know a task is meant for two teams?
- What is an actual example of when mail applies to two or more appeals streams?
- How do teams processing mail tasks determine the order in which tasks needs to be completed? 
- Which tasks can happen in parallel and which must happen serially?

**Comment by laurjpeterson** [comment]
This was an idea that I had about improving mail processes. I'm going to close it, because it isn't something we pursued.
